cli/src/commands/extensions/api.rs

```diff
@@ -223,10 +224,10 @@ async fn parse_lockfile(
 
     // Attempt to parse as requested lockfile type.
 
-    let parser = LOCKFILE_PARSERS
-        .iter()
-        .find_map(|(name, parser)| (*name == lockfile_type).then(|| *parser))
-        .ok_or_else(|| anyhow!("Unrecognized lockfile type: `{lockfile_type}`"))?;
+    let parser = lockfile_type
+        .parse::<LockFileFormat>()
+        .with_context(|| format!("Unrecognized lockfile type: `{lockfile_type}`"))?
+        .parser();
```

Comment:
I'd assume this changes the CLI's error message? Could you provide the before/after?

Reply:
It apparently does not change anything. Previously, there was no inner because find_map returns Option. The new code preserves the error from FromStr, but that information apparently doesn't survive the translation into Deno.

Before
```
Extension error: Uncaught (in promise) Error: Unrecognized lockfile type: `poetry3`
    at async Function.parseLockfile (deno:phylum:243:16)
    at async file:///Users/matt.donoughe/Documents/GitHub/cli/cli/tests/fixtures/extensions/api/main.ts:2:18
```

After
```
Extension error: Uncaught (in promise) Error: Unrecognized lockfile type: `poetry3`
    at async Function.parseLockfile (deno:phylum:243:16)
    at async file:///Users/matt.donoughe/Documents/GitHub/cli/cli/tests/fixtures/extensions/api/main.ts:2:18
```